scripts/metavar.py

```diff
@@ -1677,12 +1677,37 @@ def add_variable(self, newvar, run_env, exists_ok=False, gen_unique=False,
                                        context=newvar.context)
             # end if
         # end if
+        # Check if local_name exists in Group. If applicable, Create new 
+        # variable with uniquie name. There are two instances when new names are
+        # created:
+        # - Same <local_name> used in different DDTs.
+        # - Different <standard_name> using the same <local_name> in a Group.
+        #   During the Group analyze phase, <gen_unique> is True.
         lname = newvar.get_prop_value('local_name')
         lvar = self.find_local_name(lname)
         if lvar is not None:
+            # Check if <lvar> is part of a different DDT than <newvar>.
+            # The API uses the full variable references when calling the Group Caps,
+            # <lvar.call_string(self))> and <newvar.call_string(self)>.
+            # Within the context of a full reference, it is allowable for local_names
+            # to be the same in different data containers.
+            newvar_callstr = newvar.call_string(self)
+            lvar_callstr   = lvar.call_string(self)
+            if newvar_callstr and lvar_callstr:
+                if newvar_callstr != lvar_callstr:
+                    if not gen_unique:
+                        exists_ok = True
+                    # end if
+                # end if
+            # end if
             if gen_unique:
                 new_lname = self.new_internal_variable_name(prefix=lname)
                 newvar = newvar.clone(new_lname)
+                # Local_name needs to be the local_name for the new
```

Comment:
I am very confused by this comment. Should the first `Local_name` be `lname`?

Reply:
All instances of `lname` are replaced, there is no logic to keep the first instance. So for three instances of the same lname, we have `lname1`, `lname2`, and `lname3`, not `lname`, `lname1`, `lname2`.

Reply:
Also, this is part of #631 and does not impact the sub-cycling isssue